src/imperative/imperative.cc

```diff
@@ -494,9 +494,14 @@ std::vector<NDArray*> Imperative::Backward(
   bool prev_training = set_is_training(is_train);
```

Comment:
Do we need reset this upon exception?

Reply:
Although the lines are unrelated to the bug we are trying to fix here, it looks like we need to reset those as well.